File: src/main/java/cube8540/oauth/authentication/credentials/oauth/token/domain/OAuth2AccessTokenRepository.java
package cube8540.oauth.authentication.credentials.oauth.token.domain;

import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface OAuth2AccessTokenRepository extends JpaRepository<OAuth2AuthorizedAccessToken, OAuth2TokenId> {

    @EntityGraph(attributePaths = {"scopes", "refreshToken", "additionalInformation"})
    Optional<OAuth2AuthorizedAccessToken> findByComposeUniqueKey(OAuth2ComposeUniqueKey composeUniqueKey);

}

File: src/main/kotlin/cube8540/oauth/authentication/oauth/token/domain/OAuth2AccessTokenRepository.kt
package cube8540.oauth.authentication.oauth.token.domain

import java.util.Optional
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface OAuth2AccessTokenRepository: JpaRepository<OAuth2AuthorizedAccessToken, OAuth2TokenId> {

    @EntityGraph(attributePaths = ["scopes", "refreshToken", "additionalInformation"])
    fun findByComposeUniqueKey(composeUniqueKey: OAuth2ComposeUniqueKey): Optional<OAuth2AuthorizedAccessToken>
}